### ISSUE TITLE
Is shard weight change applied dynamically?

### ISSUE THREAD

**Issue by malikas05** [ClickHouse/ClickHouse#31306]
Hi all!
I have a question relating to the re-sharding of the cluster. We have expanded our cluster by adding one more shard. Since we want the new shard to handle more load in terms of writes, we adjusted our weights in the configs for each pertaining shard, e.g., 
`<weight>9</weight>`. Everything seems working fine; however, we wanted to know whether the restart is required for this to take into effect or is it picked up and applied dynamically? Please advise.

**Comment by den-crane** [comment]
Restart is not required.
It's picked up and applied dynamically.
You can see it in `system.clusters`

**Comment by malikas05** [comment]
@den-crane: Thanks for your response. Much appreciated!

**Comment by alexey-milovidov** [comment]
TLDR: Yes.
